lms/static/scripts/frontend_apps/utils/youtube.ts

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useState } from 'preact/hooks';
+
+import type { ThumbnailData } from '../components/URLFormWithPreview';
+import { InvalidArgumentError } from '../errors';
+
+/**
+ * Tries to resolve the video ID from a YouTube URL
+ * @return Undefined if the ID could not be resolved
+ */
+export function videoIdFromYouTubeURL(url: string): string | undefined {
```

Comment:
I note that for the JSTOR and VitalSource pickers the logic to validate a URL and extract the item ID is combined into one function. If we then need a URL representation of that ID later, we generate a URL from the item ID in a canonical format. We don't _have_ to do things that way here, though it does have the potential advantage that fewer parts of the code might need to know about all the different URL formats.

Reply:
There was two functions here because the one validating the URL throws in case of invalid URLs. Since I'm going to change that, it can probably be merged into one function.

Also, this would be another argument in favor of not throwing (see https://github.com/hypothesis/lms/pull/5359#discussion_r1185818775)